homeassistant/components/sensor/dyson.py

```diff
@@ -55,18 +69,121 @@ def should_poll(self):
         return False
 
     @property
+    def name(self):
+        """Return the name of the dyson sensor name."""
+        return self._name
+
+
+class DysonFilterLifeSensor(DysonSensor):
+    """Representation of Dyson filter life sensor (in hours)."""
+
+    def __init__(self, hass, device):
+        """Create a new Dyson filter life sensor."""
+        DysonSensor.__init__(self, hass, device)
+        self._name = "{} filter life".format(self._device.name)
+
+    @property
     def state(self):
-        """Return filter life in hours.."""
+        """Return filter life in hours."""
         if self._device.state:
-            return self._device.state.filter_life
+            return int(self._device.state.filter_life)
         return STATE_UNKNOWN
 
     @property
-    def name(self):
-        """Return the name of the dyson sensor name."""
-        return self._name
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return SENSOR_UNITS['filter_life']
+
+
+class DysonDustSensor(DysonSensor):
+    """Representation of Dyson Dust sensor (lower is better)."""
+
+    def __init__(self, hass, device):
+        """Create a new Dyson Dust sensor."""
+        DysonSensor.__init__(self, hass, device)
+        self._name = "{} dust".format(self._device.name)
+
+    @property
+    def state(self):
+        """Return Dust value."""
+        if self._device.environmental_state:
+            return self._device.environmental_state.dust
+        else:
+            return STATE_UNKNOWN
```

Comment:
No need to return `STATE_UNKNOWN`. Just return `None`. Also, no need for the `else:`.